File: entsoe-client/src/main/java/org/example/model/common/AreaType.java
package org.example.model.common;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum AreaType {
    BZN("Bidding Zone"),
    BZA("Bidding Zone Aggregation"),
    CTA("Control Area"),
    MBA("Market Balance Area"),
    IBA("Imbalance Area"),
    IPA("Imbalance Price Area"),
    LFA("Load Frequency Control Area"),
    LFB("Load Frequency Control Block"),
    REG("Region"),
    SCA("Scheduling Area"),
    SNA("Synchronous Area");

    private final String description;
}

File: entsoe_client/src/entsoe_client/model/common/area_type.py
from enum import Enum


class AreaType(Enum):
    BZN = "Bidding Zone"
    BZA = "Bidding Zone Aggregation"
    CTA = "Control Area"
    MBA = "Market Balance Area"
    IBA = "Imbalance Area"
    IPA = "Imbalance Price Area"
    LFA = "Load Frequency Control Area"
    LFB = "Load Frequency Control Block"
    REG = "Region"
    SCA = "Scheduling Area"
    SNA = "Synchronous Area"

    def __init__(self, description: str) -> None:
        self.description = description
